master/usage/routereflector/calico-routereflector.md

```diff
@@ -0,0 +1,294 @@
+allow you to remove the full mesh and scale up the size of the cluster.
+
+This guide discusses the calico/routereflector image: a container image that
+packages up the `BIRD` BGP daemon along with the `confd` templating engine to
+provide a simple route reflector image which may be used for scaled-out Calico
+deployments.
+
+The image is currently experimental due to some key limitations discussed below.
+However, it may provide a useful framework for future development.
+
+These instructions are focused around container-based deployments that use the
+calico/node container image.
+
+For an OpenStack deployment, read [Configuring BIRD as a BGP Route Reflector](bird-rr-config).
+
+#### Known limitations
+
+-  The calico/routereflector instance will automatically peer with the Calico 
+   nodes, but it currently has no mechanism to configure peerings with non-Calico
+   BGP speakers (e.g. edge routers)
+-  There is no `calicoctl` integration or similar - it is necessary to configure 
+   data directly into the `etcd` datastore each time a new route reflector 
+   is added to the deployment.
+
+## Using the Route Reflector image
+
+### Starting a Route Reflector instance
+
+On your Route Reflector host, ensure you have [Docker v1.6](http://www.docker.com) or greater
+installed.
+
+Run the following command to start the Route Reflector container image.
+
+```
+docker run -privileged -net=host -d                                \ 
+           -e IP=<IPv4_RR>                                         \
+           [-e IP6=<IPv6_RR>]                                      \
+           -e ETCD_ENDPOINTS=<http://ETCD_IP:PORT>                 \
+           calico/routereflector:{{site.data.versions[page.version].first.components["calico/routereflector"].version}}
+```
+
+Where:
+
+-  `[]` indicates an optional parameter
+-  `<IPv4_RR>` is the IPv4 address of the RR host (the BIRD instance binds to 
+   the hosts IPv4 address)
+-  `<IPv6_RR>` is the *optional* IPv6 address of the RR host (the BIRD6 instance
+   binds to the hosts IPv6 address)
+-  `<ETCD_IP:PORT>` is the colon separated IPv4 address and port of an etcd
+   node in the etcd cluster.  A comma-separated list of endpoints may be 
+   specified.
+   
+> Note: If you require TLS/SSL enabled etcd, see the [section below](#route-reflector-with-tlsssl-etcd)
+> for details on how to start the route reflector.
+
+#### Adding the Route Reflector into etcd
+
+Add an entry in etcd for this Route Reflector.  This tells the Route Reflector
+to participate in peering, and provides enough information to allow the Route
+Reflector instances to automatically form a full BGP mesh.
+
+The configuration for the Route Reflector is stored for IPv4 at:
+
+	/calico/bgp/v1/rr_v4/<RR IPv4 address>
+	
+and IPv6 at:
+
+	/calico/bgp/v1/rr_v6/<RR IPv6 address>
+
+In all cases, the data is a JSON blob in the form:
+
+        {
+          "ip": "<IP address of BGP Peer>",
```

Comment:
Isn't this just the "IP address of Route Reflector"? You haven't clearly referred to it as a "BGP peer" anywhere leading up to this